backend/src/middleware/validation/validationMiddleware.js

```diff
@@ -57,11 +57,40 @@ const validateUpdatePost = async (resolve, root, args, context, info) => {
   return validatePost(resolve, root, args, context, info)
 }
 
+const validateReport = async (resolve, root, args, context, info) => {
+  const { resourceId } = args
+  const { user, driver } = context
+  if (resourceId === user.id) throw new Error('You cannot report yourself!')
+  const session = driver.session()
+  const reportQueryRes = await session.run(
+    `
+      MATCH (:User {id:$submitterId})-[:REPORTED]->(resource {id:$resourceId}) 
+      RETURN labels(resource)[0] as label
+    `,
+    {
+      resourceId,
+      submitterId: user.id,
+    },
+  )
+  const [existingReportedResource] = reportQueryRes.records.map(record => {
+    return {
+      label: record.get('label'),
+    }
+  })
+
+  if (existingReportedResource)
+    throw new Error(
+      `You have already reported the ${existingReportedResource.label}, please only report the same ${existingReportedResource.label} once`,
+    )
+  return resolve(root, args, context, info)
+}
+
 export default {
   Mutation: {
     CreateComment: validateCommentCreation,
     UpdateComment: validateUpdateComment,
     CreatePost: validatePost,
     UpdatePost: validateUpdatePost,
+    report: validateReport,
```

Comment:
Great done @mattwr18

Reply:
glad you like it, I need to go back and do this for some of my `PR`s hahaha